server/src/test/java/io/crate/integrationtests/AlterTableRerouteIntegrationTest.java

```diff
@@ -56,4 +58,39 @@ public void testAlterTableRerouteMoveShard() throws Exception {
         execute("select * from sys.shards where id = ? and node['id'] = ? and table_name = ?", new Object[]{shardId, fromNode, tableName});
         assertThat(response).hasRowCount(0);
     }
+
+    @Test
+    public void testAlterTableRerouteMoveShardWithBkdTreeIndex() {
+        int shardId = 0;
+        String tableName = "my_table";
+        execute("create table " + tableName + " (" +
+            "id int primary key," +
+            "shape geo_shape INDEX USING bkdtree" +
+            ") clustered into 1 shards " +
+            "with (number_of_replicas=0)");
+        execute("INSERT INTO " + tableName + " (id, shape) VALUES (?, ?)", $$($(1L, "POINT (13.0 52.4)")));
+        execute("REFRESH TABLE " + tableName);
+        ensureGreen();
+        execute("select node['id'] from sys.shards where id = ? and table_name = ?", new Object[]{shardId, tableName});
+        String fromNode = (String) response.rows()[0][0];
+        execute("select id from sys.nodes where id != ?", new Object[]{fromNode});
+        String toNode = (String) response.rows()[0][0];
+
+        execute("ALTER TABLE my_table REROUTE MOVE SHARD ? FROM ? TO ?", new Object[]{shardId, fromNode, toNode});
+        assertThat(response).hasRowCount(1);
+        ensureGreen();
+        execute("select * from sys.shards where id = ? and node['id'] = ? and table_name = ?", new Object[]{shardId, toNode, tableName});
+        assertThat(response).hasRowCount(1);
+        execute("select * from sys.shards where id = ? and node['id'] = ? and table_name = ?", new Object[]{shardId, fromNode, tableName});
+        assertThat(response).hasRowCount(0);
+
+        execute("select id from " + tableName + " where match(shape, " +
+            "'POLYGON((12.998725175857544 52.40087142225922," +
+            "13.002265691757202 52.40087142225922," +
+            "13.002265691757202 52.39927416492016," +
+            "12.998725175857544 52.39927416492016," +
+            "12.998725175857544 52.40087142225922))')");
+        assertThat(response).hasRowCount(1);
+        assertThat(response).hasRows($(1));
+    }
```

Comment:
FieldMappers are used for reading from the translog. 

This test can be replaced by a test, verifying that we can read geo shapes from the translog. Something similar to https://github.com/crate/crate/commit/2cc60d723932f710ead6525e2753512304fc2d55

Reply:
We could extract parts of https://github.com/crate/crate/blob/ebfc70ae12bf1a26e88999930c3ba4235c66004c/server/src/testFixtures/java/io/crate/types/DataTypeTestCase.java#L76

into a re-usable functions where type definition/reference are parameterized.

Reply:
Thanks for the clarification. 

@mfussenegger would you mind taking a look at https://github.com/crate/crate/pull/14942/commits/2d7bab6074279f13968a60443ea03f87416454e5? I tried to extract some reusable components from the test you mentioned. Eventually, I moved the assertions that verify fields in both documents to a new Assert class and used the existing `getIndexer` method in both tests. Please let me know what you think about it.

Reply:
Looks good, thanks! 

Minor suggestion: we can make helper method in the `IndexerTest` public (slightly modify according to your change) instead of introducing a new class.

Reply:
Done. 

Previously, I introduced a separate class to avoid creating a circular dependency between the`test` and `testFixtures` directories, but perhaps it is not a problem.